src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -20,8 +24,13 @@ public DateTime StartTime
         {
             get
             {
-                // TODO: Implement this
-                throw NotImplemented.ByDesign;
+                // Get the RUsage data and convert the process start time (which is the number of
+                // nanoseconds before Now that the process started) to a DateTime.
+                DateTime now = DateTime.Now;
+                Interop.libproc.rusage_info_v3 info = Interop.libproc.proc_pid_rusage(_processId);
+                int milliseconds = Convert.ToInt32(info.ri_proc_start_abstime / NumberOfNanoSecondsInMilliseconds);
+                TimeSpan ts = new TimeSpan(0, 0, 0, 0, milliseconds);
+                return now.Subtract(ts);
```

Comment:
To avoid issues related to daylight savings and the like, you can use DateTime.UtcNow and ToLocalTime, doing the math on the UTC time rather than on the local time, e.g.
```C#
DateTime now = DateTime.UtcNow;
...
return now.Subtract(ts).ToLocalTime();
```